### PR TITLE
chore: use set to only send unique paths

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/21211

#### Type of change

- Bug fix (non-breaking change which fixes an issue)

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/workers/Evaluation/fns/utils/TriggerEmitter.ts [hunk 1]
```diff
@@ -148,9 +148,10 @@ TriggerEmitter.on(
 );
 
 export const fnInvokeLogHandler = priorityBatchedActionHandler((data) => {
+  const set = new Set([...data]);
   WorkerMessenger.ping({
     method: MAIN_THREAD_ACTION.LOG_JS_FUNCTION_EXECUTION,
-    data,
+    data: [...set],
```
Comment (arunvjn) [inline app/client/src/workers/Evaluation/fns/utils/TriggerEmitter.ts:154]: Understand the intention here, but should we do it ? If a function is run twice, we need to log it twice right ?

Comment (Rishabh-Rathod) [reply app/client/src/workers/Evaluation/fns/utils/TriggerEmitter.ts:154]: Appsmith-ee has the code below to ensure only unique function invocation are logged. Back then, a suggestion was made to have a set maintained in the worker code. Hence, this change. 

``` const funcLogged = {};

  for (const fullPath of paths) {
    const { entityName: JSObjectName, propertyPath: functionName } =
      getEntityNameAndPropertyPath(fullPath);

    if (get(funcLogged, [JSObjectName, functionName])) return;
    set(funcLogged, [JSObjectName, functionName], true);
    ```

**Review by arunvjn — APPROVED** [review]

**Review by arunvjn — APPROVED** [review]

**Pull request merged**